portfolio/src/main/webapp/script.js

```diff
@@ -47,46 +47,49 @@ function showMoreOrLessProjects() {
     
 }
 
-const ids = new Array();
-function getAndPrintComments() {
 
+const ids = new Array();
+function getAndPrintComments(){
   const commentListElement = document.getElementById('comments-container');
+  const maxComments = getMaxComments();
 
-  fetch('/data').then(response => response.json()).then((comments) => {
+  fetch('/data?maxcomments=' + maxComments).then(response => response.json()).then((comments) => {
+    if(comments.length==0) {
+        getAndSetDeleteAllButton(true);
+    }
+    else {
+        getAndSetDeleteAllButton(false);
+    }
     comments.forEach((comment) => {
       commentListElement.appendChild(createCommentElement(comment));
       ids.push(comment.id);
     })
   });
+}
 
+function getAndSetDeleteAllButton(noCommentsFlag){
     document.getElementById("delete-all-button").addEventListener("click", function(){
         for (const commentId of ids)
             deleteTask(commentId);
     });
-    /**Need to work on the on and off buttons*/
-    //const innerTextOfComments = document.getElementById("comments-container").innerText;
-    //console.log(innerTextOfComments);
-    //if(document.getElementById('comments-container').getElementsByTagName('li').length >= 1)
-    //console.log('has li in it');
-    //console.log(document.getElementById('comments-container').getElementsByTagName('li').length );
-    if(innerTextOfComments.length==0) {
+
+    if(noCommentsFlag == false) {
+        document.getElementById("delete-all-button").style.visibility="visible";
+        document.getElementById("delete-all-button").style.display="inline-block";
+        document.getElementById("commentSettings").style.display="inline-block";
+        document.getElementById("commentSettings").style.visibility="visible";
```

Comment:
You don't need to toggle the `visibility` style here. It is sufficient to use `display: none` and `display: inline-block` to hide and show elelemts.

Reply:
Hmm, I thought that they are used together?

Reply:
`display: none` behaves as if the element is not in the dom.
`visibility: hidden` hides the element but the element still takes up space on the page.

So `display: none` is stronger than the `visibility: hidden`.

Reply:
Okay, noted!